woodwork/table_accessor.py

```diff
@@ -1167,14 +1167,14 @@ def _check_index(dataframe, index):
         raise ColumnNotPresentError(
             f"Specified index column `{index}` not found in dataframe"
         )
-    if (
-        index is not None
-        and isinstance(dataframe, pd.DataFrame)
-        and not dataframe[index].is_unique
-    ):
-        # User specifies an index that is in the dataframe but not unique
-        # Does not check for Dask as Dask does not support is_unique
-        raise IndexError("Index column must be unique")
+    if index is not None and isinstance(dataframe, pd.DataFrame):
+        if not dataframe[index].is_unique:
+            # User specifies an index that is in the dataframe but not unique
+            # Does not check for Dask as Dask does not support is_unique
+            raise IndexError("Index column must be unique")
+
+        if dataframe[index].notnull().any():
+            raise IndexError("Index contains null values")
```

Comment:
@alteryx/ml-tools we should confirm whether or not this validation should happen here since it can possibly introduce multiple breaking changes in the unit tests

Reply:
Based on a quick glance, this feels like the right place for this check to happen to me.